databricks-sdk-java/src/main/java/com/databricks/sdk/core/ApiClient.java

```diff
@@ -245,15 +304,19 @@ private Response executeInner(Request in, String path) {
       Response out = null;
 
       // Authenticate the request. Failures should not be retried.
-      in.withHeaders(config.authenticate());
+      in.withHeaders(authenticateFunc.apply(null));
 
       // Prepend host to URL only after config.authenticate().
       // This call may configure the host (e.g. in case of notebook native auth).
-      in.withUrl(config.getHost() + path);
+      in.withUrl(getHostFunc.apply(null) + path);
 
       // Set User-Agent with auth type info, which is available only
       // after the first invocation to config.authenticate()
-      String userAgent = String.format("%s auth/%s", UserAgent.asString(), config.getAuthType());
+      String authType = getAuthTypeFunc.apply(null);
+      String userAgent = UserAgent.asString();
+      if (authType != "") {
```

Comment:
If getAuthTypeFunc is null, we can jst check if that func is null

Reply:
59f49977